### ISSUE TITLE
Bluetooth LE audio issue - Android 15

### ISSUE THREAD

**Issue by mrxx0** [GrapheneOS/os-issue-tracker#4237]
Since the update to android 15 my Samsung Galaxy Buds 2 pro started to behave weirdly : audio glitches, clipping and sizzling and sometimes that lead to audio server crashing (don't know if it is related or a coincidence)

I found in the device's bluetooth settings the LE audio toggle.

By deactivating it everything is back to normal.
I am sure it started since the android 15 update.

On Pixel 8 pro.

**Comment by zdrift3r** [comment]
I encounter the same situation with the same hardware. I can solve the issue by disabling LE audio, and when I enable LE audio, the issue returns.

I do not experience this issue with other bluetooth audio devices, only the Samsung Galaxy Buds 2 pro.

**Comment by crepererum** [comment]
I also have this issue w/ the Sony WF-1000XM5. Since I've paired them w/o a special app, I cannot disable LE audio for that particular device (the Android UI doesn't seem to provide a toggle for that). As a workaround, I've used developer options > Bluetooth LE audio mode > disabled (which disables LE audio for ALL devices). Now everything works :slightly_smiling_face:

**Comment by GKasp** [comment]
Same issue here with WF-1000XM4, since one of the latest GrapheneOS updates ~2 months (currently using 2025041100 for Pixel 8), the LDAC connection does not work properly anymore with intense lags and 'stuttering'. Affected is the setting with 96kHz and LE enabled. Disabling LE Audio mode as explained above or setting to LDAC 48kHz, with LE mode Unicast and Broadcast works. However, the goal is to have the functionality again available at LE enabled with LDAC 96kHz :)

**Comment by mrxx0** [comment]
Fixed in 2025070100 (android 16)

**Comment by thePanz** [comment]
@crepererum were you able to enable them?

Pairing the WF-1000XM5 the `LDAC` option is greyed out. tested with both "disabled" and "unicast - broadcast" LE Audio.
The only available codecs are AAC and SBC

→ I did not install the Sony app

**Comment by crepererum** [comment]
After a longer testing phase, I came to the conclusion my workaround from https://github.com/GrapheneOS/os-issue-tracker/issues/4237#issuecomment-2567052393 didn't really work. However after upgrading to Android 16 the issue was also fixed for me (as https://github.com/GrapheneOS/os-issue-tracker/issues/4237#issuecomment-3027189112 suggested).

**Comment by thePanz** [comment]
Thanks for the quick reply @crepererum ! 👍 

Were you able to enable the LDAC codec with those, tho?

**Comment by crepererum** [comment]
> Were you able to enable the LDAC codec with those, tho?

Yes, within the bluetooth menu (not the developer options), the toggle for "HD audio: LDAC" is set to "true"/"yes". I do have the sony app installed though and an up-to-date firmware, not sure if that makes any difference though (I don't recall if it initially worked before I had the app).
